fastapi_cache/coder.py

```diff
@@ -55,6 +55,8 @@ class JsonCoder(Coder):
     def encode(cls, value: Any) -> str:
         if isinstance(value, JSONResponse):
             return value.body
+        if cls.decode(json.dumps(value, cls=JsonEncoder)) != value:
```

Comment:
I think we did not want to encode in json and then decode and register a different value.
You are right, there is a performance concern.
Is there any other way to avoid the issue120 bug?

Reply:
Should I change the default Coder from JsonCoder to the EmptyCoder you suggested?

Reply:
Sorry for the delay in getting back to you.
the report on the issue is wrong and the same problem occurs with RedisBackend.
(I mistakenly thought it was a problem on my redis side that was preventing data from entering redisbackend, so I minsunderstood it wouldn't happen in redisbackend.)
The same would be true for other back ends.

If so, I think it would be better to write docs to use PickleCoder when using dataclass, or to fix JsonCoder's encoding to determine if it is dataclass and return a ValueError.
How do you feel about that?